src/main/java/ratpack/zipkin/internal/ZipkinHttpClientImpl.java

```diff
@@ -110,23 +100,22 @@ public void close() {
         return request(uri, requestConfigurer.prepend(RequestSpec::post));
     }
 
-    private Action<? super RequestSpec> actionWithSpan(final Action<? super RequestSpec> action, final Span span) {
+    private Action<? super RequestSpec> actionWithSpan(final Action<? super RequestSpec> action, final AtomicReference<Span> span) {
         return action.append(request -> {
-            final MethodCapturingRequestSpec captor = new MethodCapturingRequestSpec(request);
-            injector.inject(span.context(), captor.getHeaders());
-            span.start();
+            MethodCapturingRequestSpec captor = new MethodCapturingRequestSpec(this, request, span);
             action.execute(captor);
-            // TODO cannot do handler.handleSend() as the span was created out-of-scope
-            span.name(parser.spanName(adapter, captor));
-            parser.requestTags(adapter, captor, span);
         });
     }
 
-    private void streamedResponseWithSpan(Result<StreamedResponse> response, Span span) {
+    private void streamedResponseWithSpan(Result<StreamedResponse> response, AtomicReference<Span> ref) {
+        Span span = ref.get();
+        if (span == null) return;
```

Comment:
I suspect this won't be the case unless there was an exception in the request pipeline..